### PR TITLE
Fixes for redesign thread ancestry feature

### PR DESCRIPTION
**Pull request by rayane-d**
a follow-up to https://github.com/Expensify/App/pull/38722

cc @allroundexperts 
### Fixed Issues

$ https://github.com/Expensify/App/issues/39331
$ https://github.com/Expensify/App/issues/39246
$ https://github.com/Expensify/App/issues/39301
PROPOSAL: N/A

### Tests

Test 1:
1. Click on FAB > Start chat
2. Create a group chat with Users B and C
3. Send a message in the group chat
4. Right-click on the message > Reply in the thread
5. Mention User D and invite User D from the Concierge whisper
6. As User D, navigate to the thread
8. As User D, verify that the blue "Thread" above the parent message is disabled because they are not a member of the main chat.
9. As User D, verify that the blue "Thread" above the parent message color is gray.
10. As User D, verify that the parent message is not clickable (and has a default cursor)
11. As User D, verify that the header does not contain a "From ..." link to the parent thread

Test 2:

1. Send a message
2. Right-click on the message > Reply in the thread
3. Send a reply
4. Delete the reply
5. Verify that when the reply is deleted, the gray "Replies" disappear together with the deleted reply

Test 3 (Only on Web and Desktop):

1. Send a message in chat.
2. Right-click on the message > Reply in the thread.
3. Right-click on the parent message in the thread > Edit comment.
4. Hover over the text in the edit composer.
5. Verify that the cursor is text cursor when hovering over edit composer in the thread.

Test 4 (Only on Web and Desktop):

1. Send a message
2. Right-click on the message > Reply in the thread
3. Send a message in the thread
4. Verify that you can't select and copy "Thread" and "Replies words in thread dividers.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same behavior.
### QA Steps

Same as tests.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Uploading Recording 2024-04-02 042921.mp4…

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/77965000/5472a5e3-0d72-4923-ace5-1dc22c9bfe70
</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/77965000/72f1dba2-646d-4f01-a0df-726dca2354d6

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/77965000/93a43d61-1d40-4230-9ac4-513cf4392771

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/77965000/1b8edf07-4ebe-441e-84c0-491ebc078a20

https://github.com/Expensify/App/assets/77965000/961362eb-4a9a-4d3f-96a8-158d6080d5ac

</details>

<details>
<summary>MacOS: Desktop</summary>

Uploading Recording 2024-04-02 040344.mp4…

</details>

### CODE REVIEW

**Comment by pecanoro** [comment]
@rayane-djouah I see you linked https://github.com/Expensify/App/issues/39331 here, so does that mean the blocker will get fixed in this PR as well?

**Comment by pecanoro** [comment]
Ah nice, I will add myself for review as well then so I can take a look when it's ready

**Comment by hungvu193** [comment]
Seems @allroundexperts will review this one.

**Comment by pecanoro** [comment]
@rayane-djouah You are missing the screenshots/videos

File: src/pages/home/report/ThreadDivider.tsx
Comment (pecanoro) [inline src/pages/home/report/ThreadDivider.tsx:19]:
```suggestion

    /** Whether the link is disbled */
```

File: src/components/ParentNavigationSubtitle.tsx
Comment (pecanoro) [inline src/components/ParentNavigationSubtitle.tsx:28]: Please add a comment here explaining why we need this

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Comment by dubielzyk-expensify** [comment]
@dannymcclain will hold the answer but given we used the link color to signify a link, I think we probably should. But I also don't mind it being the same color cause it does separate the `Thread` from `Replies`. But yeah, let's see what Danny says :)

**Comment by shawnborton** [comment]
Yeah, that makes sense to me but also curious for Danny's opinion.

**Comment by dannymcclain** [comment]
Yeah I guess we probably should change it if it's not actually clickable. I think we have two options:
- Use same color as "Replies" (`text-supporting`)
- Keep the blue but make it 50% opacity (this would follow our disabled button state pattern)

But also, why is it disabled? How does it become disabled? Will it become un-disabled at some point? I think these questions are key to knowing which of the above choices is best.

@Expensify/design what do you think?

**Comment by shawnborton** [comment]
I prefer the idea of just "unlinking" the text and making it look gray. I think we try to avoid the disabled pattern if we can, especially since it looks a lot like offline patterns.

**Comment by dannymcclain** [comment]
>I prefer the idea of just "unlinking" the text and making it look gray. I think we try to avoid the disabled pattern if we can, especially since it looks a lot like offline patterns.

That works for me. 👍

**Comment by shawnborton** [comment]
One question though - how far "up the chain" do we expose parent comments to a user who only has access to a child thread? For example, if I have:

- Thread Level 1
    - Thread Level 2
        - Thread Level 3
        
And I only have access to view Thread Level 3, should I even be able to see the original comment from Thread Level 1?

**Comment by dannymcclain** [comment]
No I don't think so—I think in that situation you'd see the OG "thread message" from level 2 as the top-most ancestor in the thread, with no back links to the "higher up" messages.

**Comment by shawnborton** [comment]
Cool, that was my thinking too - thanks for confirming!

**Comment by dannymcclain** [comment]
Did we get rid of the message highlight when navigating to an ancestor message?

**Comment by dannymcclain** [comment]
When deleting, the `Replies` does go away as expected, but my messages just come back when I leave the thread and come back. Is that unrelated to this (or expected) or is something messed up?

https://github.com/Expensify/App/assets/7516624/12ff4536-b362-4679-8251-85f9210b3a97

**Comment by dannymcclain** [comment]
Trying to check Test 1 (disabled ancestor threads because of private rooms) but I keep running into the forced update (on web and desktop) and can't seem to get around it.

CleanShot 2024-04-02 at 15 17 12@2x

**Comment by dannymcclain** [comment]
>Trying to check Test 1 (disabled ancestor threads because of private rooms) but I keep running into the forced update (on web and desktop) and can't seem to get around it.

According to [this convo in Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1712091108118509), it looks like you may just need to merge main again.

**Comment by dannymcclain** [comment]
Ok this is generally super dope. I went through all the tests and everything seems to be working and looking as expected.

I had a couple little questions about the first test scenario:
- Do we need to use the `not allowed` cursor on hover for the top-most message? I feel like since we've removed the link and everything, we could just use the regular cursor the same as the "replies" messages.
- Is the hover action menu different for that post because it technically shows up in a chat that I don't have access to? Is that why I can't add emoji reactions and stuff?

cc'ing @Expensify/design to weigh in on my top bullet point.

https://github.com/Expensify/App/assets/7516624/e617d2ca-185d-4c29-a282-c841ec6b7446

**Comment by shawnborton** [comment]
Totally agree, I don't think we need the not-allowed cursor here.

**Comment by dannymcclain** [comment]
Ok sweet. @rayane-djouah once we update that `not-allowed` cursor, I think from a design perspective we'll be set! 🎉

**Comment by dannymcclain** [comment]
>Yes, This is the expected behavior.

Excellent!

>>Do we need to use the not allowed cursor on hover for the top-most message? I feel like since we've removed the link and everything, we could just use the regular cursor the same as the "replies" messages.
>
>Done!

Sorry, I should've been more clear in my initial comment. We shouldn't use the `not-allowed` cursor anywhere. We don't need it on the `Thread` text/icon because we already made it "regular" text instead of a link. So it should look/behave exactly like the `Replies` text/icon.

**Review by dannymcclain — DISMISSED** [review]
From a design perspective this is solid. Nice! Make sure to still get a proper technical review from an engineer 🚀

**Comment by pecanoro** [comment]
@hungvu193 Are you reviewing this one as C+?

**Comment by hungvu193** [comment]
> @hungvu193 Are you reviewing this one as C+?

I saw @rayane-djouah tagged @allroundexperts for reviewing above.

I can review this one if needed.

**Comment by pecanoro** [comment]
Ahh I see, this was a regression of https://github.com/Expensify/App/pull/38722 so we need to assign the original assignees.

**Comment by pecanoro** [comment]
@allroundexperts Please take a look when you have some time!

**Comment by allroundexperts** [comment]
@rayane-djouah https://github.com/Expensify/App/issues/39301 is still not fixed. Now we're completely hiding the parent message. I think we should still display the parent message but just not allow a click to it.

https://github.com/Expensify/App/assets/30054992/52796c46-583a-4e5d-81a0-8cd5f2b06663

**Comment by allroundexperts** [comment]
Looks solid otherwise.

**Comment by allroundexperts** [comment]
@rayane-djouah Why have we mentioned https://github.com/Expensify/App/issues/39246 in the issue description as a fixed issue then?

**Comment by allroundexperts** [comment]
Okay, lets reset. Is #39331 fixed in this PR?

**Comment by allroundexperts** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/30054992/28c52345-3931-40fb-a109-191cb5ca5d87

</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/Expensify/App/assets/30054992/a1907862-65f3-4bd9-8438-8e6f4345ece9

</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/30054992/839f8841-b46c-43e0-93ae-d4b012a47a48

</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/30054992/d352e57d-46d7-494e-a363-dc5bb4f2ff16

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/30054992/9ce1a509-8ecf-4423-ad28-56173085b3bf

</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/Expensify/App/assets/30054992/73dcd9fa-b4c1-4e67-8ce5-4f1bf5cbf80f

</details>

**Review by pecanoro — APPROVED** [review]
Code looks good! Waiting for C+ review before doing a quick test myself!

**Comment by allroundexperts** [comment]
Verified the last bug. It seems to work well!

https://github.com/Expensify/App/assets/30054992/d49f607b-9f52-4031-b7b3-9d3de17f430b

**Review by allroundexperts — APPROVED** [review]
Looks good!

**Comment by pecanoro** [comment]
@thienlnam All yours!

**Review by thienlnam — APPROVED** [review]
Look good to me, thanks!

**Pull request merged**
